### DISCUSSION TITLE
Can you control pulseeffects from another program?

### DISCUSSION THREAD

**Discussion by bcl** [wwmm/easyeffects#916] [General]
Is is possible to control the plugin settings from another program, eg. a python script?

**Comment by wwmm** [comment]
Yes. Any program capable of using the gsettings database can change our keys. Dconf is one example
![Screenshot from 2021-03-09 13-17-04](https://user-images.githubusercontent.com/16225172/110502391-ebf6a300-80d9-11eb-89d7-f233c63a7152.png)

But it can be done from the command line too through the `gsettings` command. Outdated examples in this link https://github.com/wwmm/pulseeffects/blob/master/.circleci/enable_all_plugins.sh

You can also use the GSettings API in a Python script. In our legacy Python branch from the time we used Python instead of C++ you will see many examples like this one https://github.com/wwmm/pulseeffects/blob/6c519899118e4117425193d684cb30e219fb4863/PulseEffects/bass_enhancer_presets.py#L44

**Reply by bcl** [reply]
Thanks!

**Reply by skwysham** [reply]
I’ve been trying to change EE equalizer plugin band gain settings by using either dconf and gsettings, but I have never observed changes on the equalizer GUI, aside from adding and removing the equalizer plugin, although the dconf database has the changes, e.g. **gsettings set com.github.wwmm.easyeffects.equalizer.channel:/com/github/wwmm/easyeffects/equalizer/0/leftchannel/band1-frequency 88** will return **88* with the corresponding get command.  But, looking at the equalizer GUI, the band 1 frequence remains at 64.
So, my questions are: 1) Is there some sync that I need to execute to make the equalizer gui show the new values, and 2) does the equalizer plugin ‘obey’ what’s in dconf versus what the GUI displays?

**Reply by wwmm** [reply]
>So, my questions are: 1) Is there some sync that I need to execute to make the equalizer gui show the new values, and 

Manual synchronization is not necessary. It feels more like something wrong happening in GSettings. For example when changing the values in `dconf-editor` the values should change immediately in EasyEffects Window.

>2) does the equalizer plugin ‘obey’ what’s in dconf versus what the GUI displays?

The window controls and the equalizer backend doing the digital signal processing receive the database changes in the same way. Editing the database should propagate the new values to both sides. If one is not receiving the other probably isn't too.

**Reply by skwysham** [reply]
Thanks for your response.
